src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -0,0 +1,124 @@
+#include "effects/backends/builtin/pitchshifteffect.h"
+
+#include "util/sample.h"
+
+PitchShiftGroupState::PitchShiftGroupState(
+        const mixxx::EngineParameters& engineParameters)
+        : EffectState(engineParameters) {
+    initializeBuffer(engineParameters);
+    audioParametersChanged(engineParameters);
+}
+
+PitchShiftGroupState::~PitchShiftGroupState() {
+    SampleUtil::free(m_retrieveBuffer[0]);
+    SampleUtil::free(m_retrieveBuffer[1]);
+
+    VERIFY_OR_DEBUG_ASSERT(m_pRubberBand) {
+        return;
+    }
+
+    m_pRubberBand->reset();
```

Comment:
This is redundant, because it will be called anyway invisibility. No need to care abeut the whole thing of an unique_ptr.

Reply:
Can you elaborate? This calls `RubberBandStretcher::reset()` not, `unique_ptr::reset()`... `RubberBandStretcher::reset()` will not be called implicitly, it still might be redundant though since we're in the dtor anyways.

Reply:
Oh, yes I have actually misread the code as `m_pRubberBand.reset()` sorry. 

But you assumption that it is redundant anyway is also correct. It all ends here 
https://github.com/breakfastquay/rubberband/blob/bad529f81e8ae66bd4535a6af392efa38c7fc6b1/src/StretcherImpl.cpp#L222 with almost the same code in the destructor.